### PR TITLE
Nightly Score & Finance screen

### PR DESCRIPTION
**Pull request by gnegno84**
As in title
I've also included a call to an hook for LUA script that i don't think exists (but i've added it for simmetry with end of week handling)

### CODE REVIEW

File: game/state/gametime.cpp [hunk 1]
```diff
@@ -140,9 +140,32 @@ unsigned int GameTime::getWeek() const
 	return duration.days() / 7 + 1;
 }
 
+unsigned int GameTime::getFirstDayOfCurrentWeek() const
+{
+	date today = getPtime(this->ticks).date();
```
Comment (JonnyH) [inline game/state/gametime.cpp:145]: I think these local variables can be const?

File: game/state/gametime.cpp [hunk 2]
```diff
@@ -140,9 +140,32 @@ unsigned int GameTime::getWeek() const
...
+
+		date first_day_of_week = today - days(days_to_monday);
+		return first_day_of_week.year_month_day().day;
+	}
+}
+
 unsigned int GameTime::getLastDayOfCurrentWeek() const
 {
-	unsigned int daysBeforeWeekEnd = 7 - getPtime(this->ticks).date().day_of_week();
+	unsigned short dayOfWeek = getPtime(this->ticks).date().day_of_week();
```
Comment (JonnyH) [inline game/state/gametime.cpp:164]: More local variables can be const?

**Review by JonnyH — APPROVED** [review]
Generally LGTM - though I tend to prefer every variable that can be marked as "const" to be marked as such - I think it's easier to "know" that nothing in that function modifies that variable later.

**Comment by gnegno84** [comment]
> Generally LGTM - though I tend to prefer every variable that can be marked as "const" to be marked as such - I think it's easier to "know" that nothing in that function modifies that variable later.

Thanks for the hints and checks. Totally agree with your points. I’ll update the PR asap.

**Comment by FilmBoy84** [comment]
LGTM, if the build error is solely down to Appveyor being a pain (which it appears from the log that it is), I'm good to merge if JonnyH is?

I've found no issue when building myself

**Comment by JonnyH** [comment]
Looks like the build issue was just a transitory failure - maybe due to chocolatey updating nsis or something. Seems fine after being re-kicked.

**Pull request merged**